src/NServiceBus.AzureStoragePersistence/AzureStoragePersistence.cs

```diff
@@ -12,4 +14,37 @@ internal AzureStoragePersistence()
             Supports<StorageType.Subscriptions>(s => s.EnableFeatureByDefault<AzureStorageSubscriptionPersistence>());
         }
     }
+
+    // Use this to disable synchronization storage by providing an empty implementation
+    // so that when the container tries to resolve an instance if ISynchronizedStorage it
+    // doesn't throw a ComponentNotRegisteredException exception
+    // This is hacky and we need a proper way around this.
+    public class FakeSynchronizationStorage : Feature
```

Comment:
[PR already submitted to core](https://github.com/Particular/NServiceBus/pull/3594), waiting for merging.

Unfortunately we need this in ours until that PR gets accepted into Core.